src/WarcraftLegacies.Source/Quests/BlackEmpire/QuestWakingDream.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using MacroTools;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using MacroTools.LegendSystem;
+using MacroTools.ObjectiveSystem.Objectives.LegendBased;
+using MacroTools.QuestSystem;
+using WCSharp.Shared.Data;
+
+namespace WarcraftLegacies.Source.Quests.BlackEmpire
+{
+  public sealed class QuestWakingDream : QuestData
+  {
+    private readonly LegendaryHero _zaqul;
+
+    public QuestWakingDream(LegendaryHero zaqul) : base("Waking Dream",
+      "Countless ages ago, Warlord Zon'ozz was one of the strongest generals of N'zoth. In this new time of need, he needs to be called to Azeroth once more to wage war.",
```

Comment:
Need to somehow explain how there is a connection between the Emerald Dream and Warlord Zon'ozz